core/src/main/java/org/apache/calcite/rel/rel2sql/PivotRelToSqlUtil.java

```diff
@@ -138,18 +162,35 @@ private SqlNodeList getAxisSqlNodes(Aggregate e) {
     return axesNodeList;
   }
 
-  private SqlNodeList getAggSqlNodes(Aggregate e, List<SqlNode> selectColumnList) {
+  private SqlNodeList getAxisNodeList(List<SqlNode> selectColumnList, boolean checksubquery) {
+
     final Set<SqlNode> selectList = new HashSet<>();
```

Comment:
not written by you, but i think this can be renamed as just nodeList instead of selectList

Reply:
i have renamed to 'selectNodeList'

Reply:
i suggested nodeList instead of selectlist , this methid was just returning SqlNOdelist of axisNames. Correct it in next commit.

Reply:
done